packages/datastore/__tests__/connectivityHandling.test.ts

```diff
@@ -891,6 +891,92 @@ describe('DataStore sync engine', () => {
 			const jitter: number = 0;
 			const latency: number = 1000;
 
+			/**
+			 * Helper function to perform consecutive updates on a record given it's `id`.
+			 * As explained in detail below, config options are important here, as they will affect
+			 * whether or not the outbox will merge updates.
+			 * @param originalPostId `id` of the record to update
+			 * @param numberOfUpdates number of primary client updates to perform (excludes external client updates)
+			 * @param waitOnOutbox whether or not to wait for the outbox to be empty after each update
+			 * @param pauseBeforeMutation whether or not to pause prior to the mutation
+			 */
+			const performConsecutiveUpdates = async (
+				originalPostId: string,
+				numberOfUpdates: number,
+				waitOnOutbox: boolean,
+				pauseBeforeMutation: boolean
+			) => {
+				// Mutate the original record multiple times:
+				for (let number = 0; number < numberOfUpdates; number++) {
+					/**
+					 * When we want to test a scenario where the outbox does not
+					 * necessarily merge all outgoing requests (for instance, when
+					 * we do not add artifical latency to the connection), we make
+					 * the mutations rapidly in this loop. However, because of how
+					 * rapidly this loop executes, we are creating an artifical situation
+					 * where mutations will always be merged. Setting `pauseBeforeMutation`
+					 * to `true` will adding a semi-realistic pause ("button clicks")
+					 * between updates.
+					 * Not required when awaiting the outbox after each mutation.
+					 * Additionally not required if we are intentionally testing
+					 * rapid updates, such as when the initial save is still pending.
+					 */
+					if (pauseBeforeMutation) {
+						await pause(200);
+					}
+
+					const retrieved = await DataStore.query(Post, originalPostId);
+
+					await DataStore.save(
+						// @ts-ignore
+						Post.copyOf(retrieved, updated => {
+							updated.title = `post title ${number}`;
+						})
+					);
+
+					/**
+					 * When we want to test a scenario where the user is waiting for a long
+					 * period of time between each mutation (non-concurrent updates), we wait
+					 * for an empty outbox after each mutation. This ensures each mutation
+					 * completes a full cycle before the next mutation begins. This guarantees
+					 * that there will NEVER be concurrent updates being processed by the outbox.
+					 * For use with variable latencies.
+					 */
+					if (waitOnOutbox) {
+						await waitForEmptyOutbox();
+					}
+				}
+			};
+
+			/**
+			 *
+			 * @param originalPostId `id` of the record that was updated
+			 * @param expectedFinalVersion expected final `_version` of the record after all updates are complete
+			 * @param expectedFinalTitle expected final `title` of the record after all updates are complete
+			 */
+			const commonAssertions = async (
+				originalPostId: string,
+				expectedFinalVersion: number,
+				expectedFinalTitle: string
+			) => {
```

Comment:
A more descriptive name here might help readability in the tests. Something like:

```suggestion
			const expectFinalRecordsToMatch = async (
				postId: string,
				version: number,
				title: string
			) => {
```